### PR TITLE
added document about tradeoffs between key import and portability

### LINKED ISSUE #114: Write doc about trade-offs around key portability and key import

### PR DESCRIPTION
**Pull request by michaelkernaghan**
closes #114 

Discuss the trade-offs between key import and key portability

### CODE REVIEW

**Review by AbineshECAD — COMMENTED** [review]
Empty doc file? Am I missing something?

**Review by stephengaudet — CHANGES_REQUESTED** [review]
please don't make PR for empty files

**Comment by michaelkernaghan** [comment]
the PR is not supposed to be empty; the file was not uploaded. Fixed.

File: docs/key_portability.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+## Complexity
+
+Key portability can add complexity to a system, as it may require additional steps and processes to ensure that keys are properly moved between systems. Key import can also add complexity, as it may require additional security measures and validation steps to ensure the imported keys are legitimate.
+
+## Use Cases
+
+It's important to consider the specific use cases for key portability and key import. In some cases, key portability may be more important, such as in disaster recovery scenarios where the ability to quickly recover keys is critical. In other cases, key import may be more important, such as when working with third-party systems or when importing keys for a specific purpose.
+
+# Examples of Trade-offs between Key Portability and Key Import
```
Comment (stephengaudet) [inline docs/key_portability.md:32]: I think this page is misguided.  The page is discussing the trade offs between key portability and key import.  As I understand it, key import is a feature that _supports_ key portability.  The trade off is key portability and key security.  At least, this is how I understand it. So, to me, I have trouble approving this page.

**Review by stephengaudet — CHANGES_REQUESTED** [review]

File: docs/key_portability.md (hunk 1 above)
Comment (michaelkernaghan) [reply docs/key_portability.md]: Rewritten to take this observation into account.

File: docs/key_portability.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Compatibility
+
+When it comes to key portability, compatibility is an important consideration. Keys should be compatible between different systems and devices and should be stored in a format that is widely recognized and accepted. Key import may require additional steps to ensure compatibility, such as converting keys to a different format or ensuring that the importing system supports the same cryptographic algorithms as the exporting system.
+
+## Complexity
+
+Key portability can add complexity to a system, as it may require additional steps and processes to ensure that keys are properly moved between systems. Key import can also add complexity, as it may require additional security measures and validation steps to ensure the imported keys are legitimate.
+
+## Use Cases
```
Comment (stephengaudet) [inline docs/key_portability.md:21]: it's all quite vague. I'm not sure what to suggest. What problem is this document trying to solve?

**Review by stephengaudet — CHANGES_REQUESTED** [review]

File: docs/key_portability.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+If the company prioritizes key import, it may provide third-party service access to its keys. This may make it easier to integrate the service, but it also increases the risk of the keys being compromised if the third-party service experiences a security breach.
+
+If the company prioritizes security, it may choose to keep the keys separate from the third-party service and instead provide the service with access to the encrypted data only. This makes it more difficult to integrate the service, but it ensures that the keys remain protected.
+
+## Example 3: Disaster Recovery
+
+A company wants to ensure that it can quickly recover its keys during a disaster. The company needs to balance the need for key portability with the need for security.
+
+If the company prioritizes key portability, it may choose to store its keys in a format that is widely recognized and accepted, and it may use a cloud-based service that allows them to quickly recover its keys in the event of a disaster. However, this may come at the cost of security, as the keys may be more vulnerable to theft or interception during transport.
```
[the remark this answers is not in the document]
Comment (AbineshECAD) [reply docs/key_portability.md:49]: I think "its" is the right possessive pronoun here.

[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply docs/key_portability.md:49]: no change is needed on this one

File: docs/key_portability.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# Trade-offs around Key Portability and Key Import
+
+This page discusses the trade-offs between key portability and key security. Key import is a feature that supports key portability.
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply docs/key_portability.md:3]: fixed

**Review by AbineshECAD — COMMENTED** [review]
How do users browse to this doc?

**Comment by michaelkernaghan** [comment]
> How do users browse to this doc?

i forgot the sidebar entry. I have added it and pushed the change. thanks.

**Comment by michaelkernaghan** [comment]
not required

**Pull request closed without merging**